diff --git a/src/rust/crates/example/examples/demo.rs b/src/rust/crates/example/examples/demo.rs
new file mode 100644
index 0000000..a7b5f7f
--- /dev/null
+++ b/src/rust/crates/example/examples/demo.rs
@@ -0,0 +1,2 @@
+pub fn open() {}
+fn main() {}
diff --git a/src/rust/crates/example/src/lib.rs b/src/rust/crates/example/src/lib.rs
index 14ed0d4..1ec9976 100644
--- a/src/rust/crates/example/src/lib.rs
+++ b/src/rust/crates/example/src/lib.rs
@@ -1,2 +1 @@
-pub fn open() {}
 pub fn keep() {}
